From a23d83d6c5c88bd685ab3cbf171fde245484ae6d Mon Sep 17 00:00:00 2001
From: Jonas Witschel <diabonas@archlinux.org>
Date: Tue, 7 Jul 2020 15:54:25 +0200
Subject: [PATCH] Improve reproducibility of generated JARs

maven-bundle-plugin adds a couple of extra manifest headers like
"Bnd-LastModified" recording information about the build. This makes it harder
to reproduce the JAR bit by bit on a different machine. The documented way to
remove these is the "-noextraheaders: true" instruction.

Even after applying this instruction, the ordering of the "Include-Resource"
header appears to be file system dependent. Remove this header as well to allow
for reproducible builds.
---
 pom.xml | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/pom.xml b/pom.xml
index 99a6a791e..0e53e3a78 100644
--- a/pom.xml
+++ b/pom.xml
@@ -772,6 +772,16 @@
           <excludeFilterFile>${basedir}/spotbugs-exclude-filter.xml</excludeFilterFile>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.felix</groupId>
+        <artifactId>maven-bundle-plugin</artifactId>
+        <configuration>
+          <instructions>
+            <_noextraheaders>true</_noextraheaders>
+            <_removeheaders>Include-Resource</_removeheaders>
+          </instructions>
+        </configuration>
+      </plugin>
     </plugins>
 
   </build>
-- 
2.27.0

